doc/MANUAL.adoc

```diff
@@ -314,10 +314,28 @@ this:
    `$HOME/.ccache/ccache.conf`.
 5. Otherwise, if `XDG_CONFIG_HOME` is set then use
    `$XDG_CONFIG_HOME/ccache/ccache.conf`.
-6. Otherwise, use `%APPDATA%/ccache/ccache.conf` (Windows),
+6. Otherwise, use
    `$HOME/Library/Preferences/ccache/ccache.conf` (macOS) or
    `$HOME/.config/ccache/ccache.conf` (other systems).
 
+On Windows, this is the method used:
+
+1. If `CCACHE_CONFIGPATH` is set, use that path.
+2. Otherwise, if the environment variable `CCACHE_DIR` is set then use
+   `$CCACHE_DIR/ccache.conf`.
+3. Otherwise, if <<config_cache_dir,*cache_dir*>> is set in the secondary
+   (system-wide) configuration file then use `<cache_dir>\ccache.conf`. The
+   system-wide configuration on Windows is
+   `$env:ALLUSERSPROFILE\ccache\ccache.conf` by default, `$env:ALLUSERSPROFILE`
+   is usually `C:\ProgramData`.
+4. Otherwise, if there is a legacy `$env:USERPROFILE\.ccache` directory then use
+   `$env:USERPROFILE\.ccache\ccache.conf`.
+6. Otherwise, if `$env:LOCALAPPDATA\ccache\ccache.conf` exists it is used.
+7. Otherwise, if `$env:APPDATA\ccache\ccache.conf` exists it is used.
+
+The default cache dir on Windows is `$env:LOCALAPPDATA\ccache` unless specified
+otherwise by the config. If a cache is detected in `$env:APPDATA\ccache`, which
+was the default previously, a fatal error is thrown.
```

Comment:
I'm not comfortable with potentially breaking people's builds like that. I suggest either just using the legacy location if present.

Reply:
This is what we agreed to do in #946, the issue is that `APPDATA` can be stored on the network and large caches would involve large file transfers when switching between computers in a domain for example, and since ccache cannot print a warning, a fatal error is the only option. But I will do whatever you suggest.

Reply:
But yes, this will break everyone's build if they previously used ccache in powershell or cmd with the defaults, until they manually move their cache to the `LOCALAPPDATA` directory.

Reply:
> This is what we agreed to do in https://github.com/ccache/ccache/issues/946

Right. Sorry for my short memory and for apparently changing my mind.

> since ccache cannot print a warning, a fatal error is the only option.

For the record: another option is to not even look in `APPDATA` and just leave any cache there. Do you think that would be better?

Reply:
I could remove the check for the cache in `APPDATA`, but still check for the existence of a `ccache.conf` there after `LOCALAPPDATA` in case a user wants a roaming ccache configuration, the default cache location would still be in `LOCALAPPDATA`.

Furthermore, we could add a paragraph somewhere in the appropriate place in the documentation asking users to check for cache directories in `APPDATA` and remove them if found.

Reply:
I removed the error and expanded the documentation section to include this paragraph as well as cache directory determining heuristics.